src/main/java/com/minecolonies/client/gui/WindowHutGuardTower.java

```diff
@@ -0,0 +1,467 @@
+            }
+        }
+
+        buttonJob.setEnabled(assignManually);
+
+        String auto = LanguageHandler.format("com.minecolonies.gui.workerHuts.modeA");
+        String manual = LanguageHandler.format("com.minecolonies.gui.workerHuts.modeM");
+
+        String on = LanguageHandler.format("com.minecolonies.gui.workerHuts.retrieveOn");
+        String off = LanguageHandler.format("com.minecolonies.gui.workerHuts.retrieveOn");
+
+        this.findPaneOfTypeByID(BUTTON_ASSIGNMENT_MODE, Button.class).setLabel(assignManually ? manual : auto);
+        this.findPaneOfTypeByID(BUTTON_PATROL_MODE, Button.class).setLabel(patrolManually ? manual : auto);
+        this.findPaneOfTypeByID(BUTTON_RETRIEVAL_MODE, Button.class).setLabel(retrieveOnLowHealth ? on : off);
+
+        if(task.equals(BuildingGuardTower.Task.PATROL))
+        {
+            if(patrolManually)
+            {
+                buttonSetTarget.setEnabled(true);
+                buttonSetTarget.setLabel(LanguageHandler.format("com.minecolonies.gui.workerHuts.targetPatrol"));
+            }
+            else
+            {
+                buttonSetTarget.setEnabled(false);
+            }
+            buttonTaskPatrol.setEnabled(false);
+        }
+        else if(task.equals(BuildingGuardTower.Task.FOLLOW))
+        {
+            buttonTaskFollow.setEnabled(false);
+            buttonSetTarget.hide();
+        }
+        else
+        {
+            buttonSetTarget.setLabel(LanguageHandler.format("com.minecolonies.gui.workerHuts.targetGuard"));
+            buttonTaskGuard.setEnabled(true);
+        }
+    }
+
+    /**
+     * Switch between the different task (Patrol, Follow, Guard).
+     * @param button the button clicked to switch the task.
+     */
+    private void switchTask(final Button button)
+    {
+        if(button.getID().contains("patrol"))
+        {
+            building.task = BuildingGuardTower.Task.PATROL;
+
+            buttonTaskPatrol.setEnabled(false);
+            buttonTaskFollow.setEnabled(true);
+            buttonTaskGuard.setEnabled(true);
+
+            buttonSetTarget.show();
+        }
+        else if(button.getID().contains("follow"))
+        {
+            building.task = BuildingGuardTower.Task.FOLLOW;
+
+            buttonTaskFollow.setEnabled(false);
+            buttonTaskPatrol.setEnabled(true);
+            buttonTaskGuard.setEnabled(true);
+
+            buttonSetTarget.hide();
+        }
+        else
+        {
+            building.task = BuildingGuardTower.Task.GUARD;
+
+            buttonTaskGuard.setEnabled(false);
+            buttonTaskPatrol.setEnabled(true);
+            buttonTaskFollow.setEnabled(true);
+
+            buttonSetTarget.show();
+        }
+        pullInfoFromHut();
+        sendChangesToServer();
+    }
+
+    /**
+     * Sets the target for patrolling or guarding of the guard.
+     * @param button clicked button.
+     */
+    private void setTarget(final Button button)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "button". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1172)